velox/dwio/parquet/reader/TimestampColumnReader.h

```diff
@@ -93,14 +176,13 @@ class TimestampColumnReader : public IntegerColumnReader {
       const RowSet& rows,
       ExtractValues extractValues) {
     if (auto* range = dynamic_cast<common::TimestampRange*>(filter)) {
-      // Convert TimestampRange to ParquetInt96TimestampRange.
-      ParquetInt96TimestampRange newRange = ParquetInt96TimestampRange(
-          range->lower(), range->upper(), range->nullAllowed());
+      ParquetTimestampRange<T> newRange{
+          range->lower(), range->upper(), range->nullAllowed(), timestampUnit_};
       this->readWithVisitor(
           rows,
           dwio::common::ColumnVisitor<
               int128_t,
```

Comment:
Wondering if it makes sense to use 'int64_t' for ColumnVisitor when T is int64_t. In this way the int64 timestamps don't need to be read and filtered as int128 values.

Reply:
By doing this, we may avoid allocating the int128_t vector for the int64_t timestamps.

Reply:
We still have to cast it into a `int128_t` vector, as it stores velox Timestamp typed value as 128-bit.

Reply:
Then it becomes when we do the vector expansion, before filter, or after.

Reply:
I understand your point. 'getValues' will convert the int64_t or int128_t as Timestamp, so we need to allocate the vector with a 16-byte type. Thanks!